compiler/main/config.cpp

```diff
@@ -36,18 +41,30 @@ bool                           mainPreserveDelimiter;
 void checkConfigs() {
   if (fMinimalModules == false) {
     bool             anyBadConfigParams = false;
-    Vec<const char*> configParamSetNames;
+    if (fDynoCompilerLibrary) {
+      // check that all config vars that were set from the command line were assigned
+      auto configs = parsing::configSettings(gContext);
+      for (auto config : configs) {
+        auto usedId = parsing::nameToConfigSettingId(gContext, config.first);
```

Comment:
I was originally going to advocate that we make both the getter and the setter file-private, but I didn't because Ahmad calls the getter in `checkConfigs()`. Which means we'd need to write the whole config check thing in dyno as well. Which might be fine?

I still think we need to address the getter and setter thing, though.